query_executor.go

```diff
@@ -1,22 +1,26 @@
 package gocql
 
 import (
+	"sync"
 	"time"
 )
 
 type ExecutableQuery interface {
 	execute(conn *Conn) *Iter
 	attempt(keyspace string, end, start time.Time, iter *Iter, host *HostInfo)
 	retryPolicy() RetryPolicy
+	speculativeExecutionPolicy() SpeculativeExecutionPolicy
 	GetRoutingKey() ([]byte, error)
 	Keyspace() string
 	Cancel()
+	IsIdempotent() bool
 	RetryableQuery
 }
 
 type queryExecutor struct {
 	pool   *policyConnPool
 	policy HostSelectionPolicy
+	specWG sync.WaitGroup
```

Comment:
we need one of these per query, define it local to the execution loop

Reply:
will do.